tests/models/unit_cases/case_3a_setpoint/src/run_3a.py

```diff
@@ -0,0 +1,164 @@
+import numpy as np
+
+from rtctools.optimization.collocated_integrated_optimization_problem import (
+    CollocatedIntegratedOptimizationProblem,
+)
+from rtctools.optimization.goal_programming_mixin import Goal
+from rtctools.optimization.linearized_order_goal_programming_mixin import (
+    LinearizedOrderGoalProgrammingMixin,
+)
+from rtctools.optimization.single_pass_goal_programming_mixin import SinglePassGoalProgrammingMixin
+
+from rtctools_heat_network.esdl.esdl_additional_vars_mixin import ESDLAdditionalVarsMixin
+from rtctools_heat_network.esdl.esdl_mixin import ESDLMixin
+from rtctools_heat_network.techno_economic_mixin import TechnoEconomicMixin
+
+
+class TargetDemandGoal(Goal):
+    priority = 1
+
+    order = 2
+
+    def __init__(self, state, target):
+        self.state = state
+
+        self.target_min = target
+        self.target_max = target
+        self.function_range = (0.0, 2.0 * max(target.values))
+        self.function_nominal = np.median(target.values)
+
+    def function(self, optimization_problem, ensemble_member):
+        return optimization_problem.state(self.state)
+
+
+class ConstantGeothermalSource(Goal):
+    priority = 2
+
+    order = 2
+
+    def __init__(self, optimization_problem, source, target, lower_fac=0.9, upper_fac=1.1):
+        self.target_min = lower_fac * target
+        self.target_max = upper_fac * target
+        self.function_range = (0.0, 2.0 * target)
+        self.state = f"{source}.Q"
+        self.function_nominal = optimization_problem.variable_nominal(self.state)
+
+    def function(self, optimization_problem, ensemble_member):
+        return optimization_problem.state(self.state)
+
+
+class MinimizeSourcesHeatGoal(Goal):
+    priority = 3
+
+    order = 1
+
+    def __init__(self, source):
+        self.target_max = 0.0
+        self.function_range = (0.0, 10e6)
+        self.source = source
+        self.function_nominal = 1e6
+
+    def function(self, optimization_problem, ensemble_member):
+        return optimization_problem.state(f"{self.source}.Heat_source")
+
+
+class MinimizeSourcesFlowGoal(Goal):
+    priority = 4
+
+    order = 2
+
+    def __init__(self, source, nominal=1.0):
+        self.target_max = 0.0
+        self.function_range = (0.0, 1.0e3)
+        self.source = source
+        self.function_nominal = nominal
+
+    def function(self, optimization_problem, ensemble_member):
+        return optimization_problem.state(f"{self.source}.Q") * 1.0e3
+
+
+class MinimizeSourcesQTHGoal(Goal):
+    priority = 3
+
+    order = 2
+
+    def __init__(self, source):
+        self.source = source
+        self.function_nominal = 1e6
+
+    def function(self, optimization_problem, ensemble_member):
+        return optimization_problem.state(f"{self.source}.Heat_source")
```

Comment:
Nor these goals are used. Or do you plan to remove the setpointconstraint checks from case_3a/src/run_3a.py ?